server/build/webpack/base.config.js

```diff
@@ -0,0 +1,318 @@
+    resolve: {
+      extensions: ['.js', '.jsx', '.json', '.ts', '.tsx'],
+      modules: [
+        nextNodeModulesDir,
+        'node_modules'
+      ],
+      alias: {
+        'next': nextDir,
+        'react-dom': dev ? 'react-dom/cjs/react-dom.development.js' : 'react-dom/cjs/react-dom.production.min.js'
+      }
+    },
+    resolveLoader: {
+      modules: [
+        nextNodeModulesDir,
+        'node_modules',
+        path.join(__dirname, '..', 'loaders')
+      ]
+    },
+    module: {
+      rules: [
+        dev && !isServer && {
+          test: /\.(js|jsx)(\?[^?]*)?$/,
+          loader: 'hot-self-accept-loader',
+          include: [
+            path.join(dir, 'pages'),
+            nextPagesDir
+          ]
+        },
+        {
+          test: /\.+(js|jsx)$/,
+          include: [dir],
+          exclude: /node_modules/,
+          use: {
+            loader: 'babel-loader',
+            options: babelLoaderOptions
+          }
+        },
+        {
+          test: /\.+(ts|tsx)$/,
+          include: [dir],
+          exclude: /node_modules/,
+          use: [
+            {
+              loader: 'babel-loader',
+              options: babelLoaderOptions
+            },
+            {
+              loader: 'ts-loader',
+              options: {
+                transpileOnly: true
+              }
+            }
+          ]
+        },
+        {
+          test: /\.css$/,
+          use: cssLoaderConfig()
+        },
+        {
+          test: /\.scss$/,
+          use: cssLoaderConfig('sass-loader')
+        },
+        {
+          test: /\.less$/,
+          use: cssLoaderConfig('less-loader')
+        }
+      ].filter(Boolean)
+    },
+    plugins: [
+      // Useful when profiling
+      // new webpack.ProgressPlugin({
+      //   profile: true
+      // }),
+      // new StatsPlugin(`stats-${isServer ? 'server':'client'}.json`),
+      new webpack.IgnorePlugin(/(precomputed)/, /node_modules.+(elliptic)/),
+      dev && new webpack.NoEmitOnErrorsPlugin(),
+      dev && !isServer && new FriendlyErrorsWebpackPlugin(),
+      dev && new webpack.NamedModulesPlugin(),
+      dev && new webpack.HotModuleReplacementPlugin(), // Hot module replacement
+      dev && new CaseSensitivePathPlugin(), // Since on macOS the filesystem is case-insensitive this will make sure your path are case-sensitive
+      dev && new webpack.LoaderOptionsPlugin({
+        options: {
+          context: dir,
+          customInterpolateName (url, name, opts) {
+            return interpolateNames.get(this.resourcePath) || url
+          }
+        }
+      }),
+      dev && new WriteFilePlugin({
+        exitOnErrors: false,
+        log: false,
+        // required not to cache removed files
+        useHashIndex: false
+      }),
+      !dev && new webpack.IgnorePlugin(/react-hot-loader/),
+      !isServer && !dev && new UglifyJSPlugin({
+        exclude: /react\.js/,
+        parallel: true,
+        sourceMap: false,
```

Comment:
May be true, if the `devtool === 'source-map'`.

Reply:
We can't detect `devtool === 'source-map'` because it'd be set in next.config.js 😅